src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -157,42 +188,52 @@ void KeyboardManagerState::UpdateDetectSingleKeyRemapUI()
     }
 
     std::unique_lock<std::mutex> detectedRemapKey_lock(detectedRemapKey_mutex);
-    hstring remapKeyString = winrt::to_hstring((unsigned int)detectedRemapKey);
+    hstring key = winrt::to_hstring((unsigned int)detectedRemapKey);
     detectedRemapKey_lock.unlock();
 
     // Since this function is invoked from the back-end thread, in order to update the UI the dispatcher must be used.
-    currentSingleKeyRemapTextBlock.Dispatcher().RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [=]() {
-        currentSingleKeyRemapTextBlock.Text(remapKeyString);
+    currentSingleKeyRemapTextBlock.Dispatcher().RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [this, key]() {
+        currentSingleKeyRemapTextBlock.Children().Clear();
+        AddKeyToLayout(currentSingleKeyRemapTextBlock, key);

```

Comment:
For the single key case should we be clearing and adding a new block each time? Once the first key is pressed it should be sufficient to just update the text on the same block right? I'm not sure about the performance impact of adding new UI controls each time. It's necessary for shortcuts I guess but it could be avoided for single key.

Reply:
I think it might be worth it to take a minor performance hit in this case to keep the handling of the multiple keys case and the single key case as similar as possible, just to keep the code simple. This could also make refactoring into a XAML based UserControl easier in the future, if thats going to happen.